script.js

```diff
@@ -27,127 +27,368 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 
 my_widget_script =
-{
-  init:function (mode, json_data) {
-    //this method is called when the form is being constructed
-    // parameters
-    // mode = if it equals 'view' than it should not be editable
-    //        if it equals 'edit' then it will be used for entry
-    //        if it equals 'view_dev' same as view,  does some additional checks that may slow things down in production
-    //        if it equals 'edit_dev' same as edit,   does some additional checks that may slow things down in production
- 
-    // json_data will contain the data to populate the form with, it will be in the form of the data
-    // returned from a call to to_json or empty if this is a new form.
-    //By default it calls the parent_class's init.
- 
-    //TO DO write code specific to your form
-    this.parent_class.init(mode, json_data);
-    
-    if (mode.indexOf('view') > -1) {
-      var isEmpty = function(tr) {
-        var inputs = $('input', tr);
-        for (var i = 0; i < inputs.length; i++) {
-          if ($(inputs[i]).val()) {
-            return false;
+    {
```

Comment:
It seems strange that this is indented by 4 spaces. I'd expect either 0 or 2 when comparing to the rest of the file. The increase in indentation has made it much harder to review the diff since *every line appears to have changed* when only some have.